internal/services/machinelearning/machine_learning_datastore_blobstorage_resource.go

```diff
@@ -0,0 +1,268 @@
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceMachineLearningDataStore() *pluginsdk.Resource {
+	resource := &pluginsdk.Resource{
+		Create: resourceMachineLearningDataStoreCreateOrUpdate,
+		Read:   resourceMachineLearningDataStoreRead,
+		Update: resourceMachineLearningDataStoreCreateOrUpdate,
+		Delete: resourceMachineLearningDataStoreDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := datastore.ParseDataStoreID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DataStoreName,
+			},
+
+			"workspace_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.WorkspaceID,
+			},
+
+			"storage_account_name": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validation.StringIsNotEmpty,
+			},
+
+			"container_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"is_default": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"service_data_auth_identity": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(datastore.ServiceDataAccessAuthIdentityNone),
+					string(datastore.ServiceDataAccessAuthIdentityWorkspaceSystemAssignedIdentity),
+					string(datastore.ServiceDataAccessAuthIdentityWorkspaceUserAssignedIdentity),
+				},
+					false),
+				Default: string(datastore.ServiceDataAccessAuthIdentityNone),
+			},
+
+			"account_key": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				ExactlyOneOf: []string{"account_key", "shared_access_signature"},
+			},
+
+			"shared_access_signature": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				AtLeastOneOf: []string{"account_key", "shared_access_signature"},
+			},
+
+			"tags": commonschema.TagsForceNew(),
+		},
+	}
+	return resource
+}
+
+func resourceMachineLearningDataStoreCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
```

Comment:
This needs to be split into separate create and update methods.

Reply:
service only have one API for create/update, do need to split it into create and update method?

Reply:
We do, so that ignore_changes works.

Reply:
updated